softwerkskammer/lib/wiki/wikiService.js

```diff
@@ -64,11 +79,13 @@ module.exports = {
     });
   },
 
+  replaceNonExistentNicknames,
+
   pageHistory: function pageHistory(completePageName, callback) {
     Git.readFile(completePageName + '.md', 'HEAD', err => {
       if (err) { return callback(err); }
       Git.log(completePageName + '.md', 'HEAD', 30, (ignoredErr, metadata) => {
-        callback(null, metadata);
+        replaceNonExistentNicknames(metadata, callback);
```

Comment:
Lässt sich das vielleicht schon beim Einlesen der metadata geradeziehen? Fände ich glaube ich natürlicher als nochmal so im Nachgang über die Daten drüberzuschleifen...

Reply:
Würde ich nicht. Sind 2 bounded contexts: Git -> Author, Member -> Nickname

Reply:
OK, verstehe. 

Ich frag mich grad ob man dieses Replacement nicht eigentlich bei jedem Aufruf von Git.log() machen müsste?

Reply:
hat aber im gitmech module nix verloren...

Reply:
und log wird nur hier aufgerufen afaik

Reply:
Nein, ich rede nicht von dem gitmech module, sondern vom wikiService.
Git.log() wird 4 x aufgerufen in wikiService, aber Du hast jetzt nur eine Stelle angepasst. Was passiert mit den anderen?

Reply:
ich guck

Reply:
Ja, NOCH nicht. Ich finde es einfach nur etwas unsauber, die Metadata nur in manchen Situationen zu bereinigen und in anderen nicht.

Wie wäre es, im wikiService eine Funktion sanitizedLog() zu haben, die Git.log + Bereinigung aufruft?